### PR TITLE
Osiris/audit review

### PR DESCRIPTION
**Pull request by 0xOsiris**
Addresses Internal Audit Review nits

### CODE REVIEW

File: contracts/src/PBHSignatureAggregator.sol [hunk 1]
```diff
@@ -82,7 +87,7 @@ contract PBHSignatureAggregator is IAggregator {
                 userOps[i].signature.length == expectedLength + PROOF_DATA_LENGTH,
                 InvalidSignatureLength(expectedLength + PROOF_DATA_LENGTH, userOps[i].signature.length)
             );
-            bytes memory proofData = userOps[i].signature[expectedLength:];
+            bytes memory proofData = userOps[i].signature[expectedLength:expectedLength + PROOF_DATA_LENGTH];
```
Comment (Dzejkop) [inline contracts/src/PBHSignatureAggregator.sol:90]: Do we have a test for this edge case?

Comment (0xOsiris) [reply contracts/src/PBHSignatureAggregator.sol:90]: Yes, we do [here](https://github.com/worldcoin/world-chain/blob/1cbf420f5b23067e3fd6b9c9a4f6d4f1172c184a/contracts/test/PBHSignatureAggregator.t.sol#L131). The require statement above will always revert before we hit this line if the end of the byte array isn't `expectedLength + PROOF_DATA_LENGTH`

Comment (0xOsiris) [reply contracts/src/PBHSignatureAggregator.sol:90]: It's identical logic. Thought it was best to be explicit though

File: contracts/src/PBH4337Module.sol [hunk 2]
```diff
@@ -8,21 +8,35 @@ import {_packValidationData} from "@account-abstraction/contracts/core/Helpers.s
...
+
+    /// @notice The PBH Nonce Key.
+    /// @dev This key is used to identify a PBH user operation.
     uint192 public immutable PBH_NONCE_KEY;
 
+    /// @notice Thrown when the proof size is invalid.
     error InvalidProofSize();
 
+    /// @notice Thrown when a null data is passed in the constructor.
+    error ZeroInitializedData();
```
Comment (0xKitsune) [inline contracts/src/PBH4337Module.sol]: Nit: Can we update this to `AddressZero()` to be consistent with the `PBHEntryPointImplV1` contract?

File: contracts/src/PBHSignatureAggregator.sol [hunk 3]
```diff
@@ -27,10 +28,14 @@ contract PBHSignatureAggregator is IAggregator {
     /// @notice Thrown when the length of the signature is invalid.
     error InvalidSignatureLength(uint256 expected, uint256 actual);
 
+    /// @notice Thrown when a zero address is passed as the PBHEntryPoint.
+    error ZeroAddress();
```
Comment (0xKitsune) [inline contracts/src/PBHSignatureAggregator.sol]: Can we either update this to `AddressZero()` or update the `PBHEntryPointImplV1` to use `ZeroAddress()` for consistency?

File: contracts/src/PBH4337Module.sol (hunk 2 above)
Comment (0xOsiris) [reply contracts/src/PBH4337Module.sol]: Do we want another error for a zero `PBH_NONCE_KEY`?

**Review by 0xKitsune — APPROVED** [review]
